lib/MPIPreferences/src/MPIPreferences.jl

```diff
@@ -75,7 +82,8 @@ function use_jll_binary(binary = Sys.iswindows() ? "MicrosoftMPI_jll" : "MPICH_j
         force=force
     )
 
-    @warn "The underlying MPI implementation has changed. You will need to restart Julia for this change to take effect" binary
+    PREFS_CHANGED[] = true
+    @info "MPIPreferences changed" binary
```

Comment:
Should we check whether the preferences actually changed, in case one just maintained the same as before?

Reply:
Yeah, I think that is worthwhile.